diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_reg.h b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_reg.h
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_reg.h
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_reg.h
@@ -7115,2 +7115,3 @@
 #define MISC_REG_CPMU_LP_MASK_EXT_4				 0xa644
+#define MISC_REG_WC0_CTRL_REG				 0xa6c0
 #define MISC_REG_XT_SPIO_INT_STATUS				 0xa4ac

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
@@ -3955,2 +3955,5 @@ static void bnx2x_warpcore_set_2500x(struct link_params *params,
 	struct bnx2x *bp = params->bp;
+	
+	REG_WR(bp, MISC_REG_WC0_CTRL_REG, 0x70);
+
 	DP(NETIF_MSG_LINK, "Setting 2500X\n");
@@ -4436,2 +4439,3 @@ static bnx2x_lstatus_t bnx2x_warpcore_read_status(struct link_params *params,
 		vars->link_status |= LINK_1000baseX;
+	} else if (val1 & (1 << 13)) {
 	} else if (val1 & (1 << 3)) {
@@ -8127,2 +8131,4 @@ static bnx2x_lstatus_t bnx2x_link_initialize(struct link_params *params,
 	bnx2x_acquire_phy_lock(params);
+	REG_WR(bp, MISC_REG_WC0_CTRL_REG, 0x70);
+	bnx2x_release_phy_lock(params);
 
@@ -12200,2 +12206,4 @@ bnx2x_lstatus_t bnx2x_phy_init(struct link_params *params, struct link_vars *var
 	bnx2x_acquire_phy_lock(params);
+	REG_WR(bp, MISC_REG_WC0_CTRL_REG, 0x70);
+	bnx2x_release_phy_lock(params);
