diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_reg.h b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_reg.h
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_reg.h
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_reg.h
@@ -7114,6 +7114,7 @@
 #define MISC_REG_CPMU_LP_MASK_EXT_3				 0xa640
 #define MISC_REG_CPMU_LP_MASK_EXT_4				 0xa644
 #define MISC_REG_CPMU_LP_SM_ENT_CNT				 0xa4b4
+#define MISC_REG_WC0_CTRL_REG					 0xa6c0
 #define MISC_REG_XT_SPIO_INT_STATUS				 0xa4ac
 
 /* [RW 32] clk_mng_mode. 0 - reset; 1 - bypass; 2 - normal. Bit 0 - core
diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
@@ -3956,6 +3956,8 @@ static void bnx2x_warpcore_set_2500x(struct link_params *params,
 	struct bnx2x *bp = params->bp;
 	u32 lane = bnx2x_get_warpcore_lane(params);
 
+	REG_WR(bp, MISC_REG_WC0_CTRL_REG, 0x70);
+
 	bnx2x_cl45_read_and_write(bp, params->port, PHY_WC_DEVAD,
 				  MDIO_WC_REG_IEEE0BLK_AUTONEGNP, 0x0100);
 
@@ -8132,6 +8134,7 @@ static bnx2x_lstatus_t bnx2x_link_initialize(struct link_params *params,
 		}
 	}
 	bnx2x_acquire_phy_lock(params);
+	REG_WR(bp, MISC_REG_WC0_CTRL_REG, 0x70);
 	bnx2x_release_phy_lock(params);
 
 	if (non_ext_phy != ext_phy) {
@@ -12205,6 +12208,7 @@ bnx2x_lstatus_t bnx2x_phy_init(struct link_params *params, struct link_vars *var
 	params->link_flags |= PHY_INITIALIZED;
 
 	bnx2x_acquire_phy_lock(params);
+	REG_WR(bp, MISC_REG_WC0_CTRL_REG, 0x70);
 	bnx2x_release_phy_lock(params);
 
 	return bnx2x_link_initialize(params, vars);
